torch/distributions/lowrank_multivariate_normal.py

```diff
@@ -102,17 +101,17 @@ def __init__(self, loc, cov_factor, cov_diag, validate_args=None):
         if cov_diag.shape[-1:] != event_shape:
             raise ValueError("cov_diag must be a batch of vectors with shape {}".format(event_shape))
 
-        scale_batch_shape = _get_batch_shape(cov_factor, cov_diag)
+        loc_ = loc.unsqueeze(-1)
+        cov_diag_ = cov_diag.unsqueeze(-1)
         try:
-            batch_shape = torch._C._infer_size(loc.shape[:-1], scale_batch_shape)
+            loc_, self.cov_factor, cov_diag_ = torch.broadcast_tensors(loc_, cov_factor, cov_diag_)
         except RuntimeError:
             raise ValueError("Incompatible batch shapes: loc {}, cov_factor {}, cov_diag {}"
                              .format(loc.shape, cov_factor.shape, cov_diag.shape))
+        self.loc = loc_[..., 0]
+        self.cov_diag = cov_diag_[..., 0]
+        batch_shape, event_shape = self.loc.shape[:-1], self.loc.shape[-1:]
```

Comment:
You're still setting `event_shape` above, so this is a bit confusing

Reply:
I think this should be the same as L94 above, in which case we can delete the `event_shape` assignment here?